src/writer/app_runner.py

```diff
@@ -644,14 +648,37 @@ def _set_logger(self):
 
     def _start_fs_observer(self):
         self.observer = PollingObserver(AppRunner.UPDATE_CHECK_INTERVAL_SECONDS)
-        self.observer.schedule(FileEventHandler(self.reload_code_from_saved), path=self.app_path, recursive=True)
+        self.observer.schedule(FileEventHandler(self.reload_code_from_saved, patterns=["*.py"]), path=self.app_path, recursive=True)
+        self.observer.schedule(FileEventHandler(self._install_requirements, patterns=["requirements.txt"]), path=self.app_path)
```

Comment:
You don't have access to the session here. If you want to install, you should do that in the request. The challenge it's it will take some time.